### PR TITLE
Bugfix/numerator obs values

### PR DESCRIPTION
**Pull request by RohitKandimalla**
Pull requests into cqm-execution require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Case:** While generating observation results for each episode, we verify if the episode is a part of Numerator or Denominator, When Episode is not a part of Denominator, then the first index of `observation_values` array for that episode should be defaulted to 0.
**Bug:** Instead of adding 0 at the first index, we are replacing the existing value, which could be the value of Numerator Observation.

[https://jira.cms.gov/browse/MAT-6623](https://jira.cms.gov/browse/MAT-6623)
Screenshot of the contents from This story 
image

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by dczulada** [comment]
@RohitKandimalla is it possible to post the contents of the JIRA issue (https://jira.cms.gov/browse/MAT-6623 ) in this pull request?  I'm having trouble understanding what this is trying to fix.

**Comment by RohitKandimalla** [comment]
> @RohitKandimalla is it possible to post the contents of the JIRA issue (https://jira.cms.gov/browse/MAT-6623 ) in this pull request? I'm having trouble understanding what this is trying to fix.

Sure I add a screenshot, so that it will easier to understand c&p the contents. The over all issue is that Numerator Observations results are not returned by cqm-execution.

Technically, cql-execution is returning us the num_obs values, but during the construction of observation_values array, we are overriding existing values instead of unshifting the array.

File: lib/helpers/calculator_helpers.js [hunk 1]
```diff
@@ -116,12 +116,14 @@ module.exports = class CalculatorHelpers {
...
         if ('observation_values' in populationResults) {
           // DENOM observation will be the first of 2 observations
+          populationResultsHandled.observation_values[1] = populationResultsHandled.observation_values[0];
           populationResultsHandled.observation_values[0] = 0;
         }
       }
       if (populationResults.DENEX != null && !this.isValueZero('DENEX', populationResults) && populationResults.DENEX >= populationResults.DENOM) {
         if ('observation_values' in populationResults) {
           // DENOM observation will be the first of 2 observations
+          populationResultsHandled.observation_values[1] = populationResultsHandled.observation_values[0];
```
Comment (dczulada) [inline lib/helpers/calculator_helpers.js:126]: I do not believe you want to upshift the observation here.

In this case (Where the patient calculates into the DENEX and DENOM) there will value for the Denominator Observation in the 0 index of the observation_values array.  If the patient happens to be in the Numerator, the Numerator Observation will already be in the 1 index of the observation_value array.

If this code change is kept, then the Denominator Observation value will incorrectly override the Numerator Observation in scenarios when a patient meets the DENOM/DENEX and NUMER criteria.

Here are some screen captures showing the results from a sample Ratio measure I've attached (as CQL)
[SampleRatio.cql.zip](https://github.com/projecttacoma/cqm-execution/files/14270159/SampleRatio.cql.zip)

**This is the original calculation.  The NUMER Observation is missing when a patient is in the NUMER and not in the DENOM.**

Original - With Bug

**This is the calculation for this PR as written.  The DENOM Observation (of 1) is overriding the NUMER Observation (of 2)**

PR - As written

**This is the calculation without this line of code (Line 126 above).  The NUMER Observation of 2 is not being overridden.**

PR - Without line 126

Comment (RohitKandimalla) [reply lib/helpers/calculator_helpers.js:126]: @dczulada I was able to reproduce this scenario, where the same episode is part of IPP, DENOM, DENEX and NUM
Working on addressing this scenario right now, I will have a commit shortly

Comment (RohitKandimalla) [reply lib/helpers/calculator_helpers.js]: Thank you so much, I reverted it and added a new test case to make sure.

Comment (adongare) [reply lib/helpers/calculator_helpers.js]: looks like looking good now. works in Bonnie.

**Review by adongare — DISMISSED** [review]

**Review by dczulada — APPROVED** [review]

**Review by adongare — APPROVED** [review]
